tests/integration/schema_registry/test_jsonschema.py

```diff
@@ -234,8 +235,14 @@ async def not_schemas_are_backward_compatible(
 
 @pytest.mark.parametrize("trail", ["", "/"])
 @pytest.mark.parametrize("compatibility", [CompatibilityModes.FORWARD, CompatibilityModes.BACKWARD, CompatibilityModes.FULL])
+@pytest.mark.parametrize("metadata", [None, {}])
+@pytest.mark.parametrize("rule_set", [None, {}])
 async def test_same_jsonschema_must_have_same_id(
-    registry_async_client: Client, compatibility: CompatibilityModes, trail: str
+    registry_async_client: Client,
+    compatibility: CompatibilityModes,
+    trail: str,
+    metadata: Dict[str, Any],
```

Comment:
**[non-blocking]**
Maybe we can create new types for these fields, i.e. using python [NewType](https://docs.python.org/3/library/typing.html#newtype), like done [here](https://github.com/Aiven-Open/karapace/blob/main/karapace/typing.py#L24).